.github/workflows/integration-test-containers.yml

```
@@ -16,15 +16,47 @@ on:
         type: boolean
         required: true
         description: Whether the QA containers should be rebuilt
+    outputs:
+      collector-tests-tag:
+        description: The tag used for the integration test image
+        value: ${{ jobs.build-test-image.outputs.collector-tests-tag || 'master' }}
 
 jobs:
+  should-build-test-image:
+    name: Determine if tests image needs to be built
+    runs-on: ubuntu-latest
+    outputs:
+      build-image: ${{ steps.changed.outputs.tests-changed }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: dorny/paths-filter@v3
+        id: changed
+        with:
+          filters: |
+            tests-changed:
+              - integration-tests/pkg/**
```

Think we are missing the suites directory here
```suggestion
              - integration-tests/pkg/**
              - integration-tests/suites/**
```

---

Damn it, I knew I thought I'd missed something